src/providers/http_provider.py

```diff
@@ -12,6 +12,10 @@
 logger = logging.getLogger(__name__)
 
 
+class NoActiveProviderError(Exception):
```

Comment:
This won't work, NoActiveProviderError in lib and here are different exceptions. This exception wont be catched in BaseOracleModule
maybe just import NoActiveProviderError from multi_http_provider?